bpo-41762: Fix usage of productionlist markup in the doc

vstinner opened the pull request:
Use an unique identifier for the different grammars documented using
the Sphinx productionlist markup.

productionlist markups of the same grammar, like "expressions" or
"compound statements", use the same identifier.

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->


<!-- issue-number: [bpo-41762](https://bugs.python.org/issue41762) -->
https://bugs.python.org/issue41762
<!-- /issue-number -->

vstinner committed:
bpo-41762: Fix usage of productionlist markup in the doc

Use an unique identifier for the different grammars documented using
the Sphinx productionlist markup.

productionlist markups of the same grammar, like "expressions" or
"compound statements", use the same identifier.

vstinner commented on Doc/reference/toplevel_components.rst:
```diff
@@ -85,7 +85,7 @@ Interactive input
 
-.. productionlist::
+.. productionlist:: interactive-input-component
    interactive_input: [`stmt_list`] NEWLINE | `compound_stmt` NEWLINE
```
This change breaks the link to stmt_list.

Should we use the same identifier here, in expressions, simple statements and compound statements?

pablogsal replied:
Yeah, I think that is a good idea (to preserve those links). Is there any other missing links between sections?

pablogsal replied:
Or maybe use the least amount of namespaces possible?

vstinner replied:
I didn't check all modified pages.

corona10 reviewed (approved):
@vstinner @pablogsal 

I can check my PR works well after this PR is merged :)
https://github.com/python/cpython/pull/22198

vstinner committed:
Use "python-grammar" identifier

compound_stmts, expressions, simple_stmts and toplevel_components now
use the "python-grammar" identifier for their "productionlist"
markups.

vstinner committed:
lexical_analysis also uses python-grammar
